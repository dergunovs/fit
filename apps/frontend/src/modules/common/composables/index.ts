import { ref, computed, onMounted } from 'vue';
import { useRouter, useRoute } from 'vue-router';

import LayoutAdmin from '@/common/components/LayoutAdmin.vue';
import LayoutDefault from '@/common/components/LayoutDefault.vue';

export function useLayout() {
  const router = useRouter();
  const route = useRoute();

  const isLoaded = ref(false);

  const layoutComponent = computed(() => (route.meta.layout === 'admin' ? LayoutAdmin : LayoutDefault));

  onMounted(async () => {
    await router.isReady();
    isLoaded.value = true;
  });

  return {
    isLoaded,
    layoutComponent,
  };
}

export function useRouteId(title: string, isQuery?: boolean) {
  const route = useRoute();

  const id = computed(() => {
    const value = isQuery ? route.query[title]?.toString() : route.params[title].toString();

    return value || '';
  });

  return { id };
}
