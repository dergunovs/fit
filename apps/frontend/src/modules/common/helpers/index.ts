import { toast } from 'mhz-ui';

export async function copyToClipboard(text: string, toastText: string) {
  await navigator.clipboard.writeText(text);

  toast.success(toastText);
}
