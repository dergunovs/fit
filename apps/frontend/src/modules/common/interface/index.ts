import { TPostAuthLoginDTO } from 'fitness-tracker-contracts';
import { FunctionalComponent } from 'vue';

export interface INavItem {
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IOnSuccess {
  create?: (id?: string) => Promise<void>;
  update?: () => Promise<void>;
  delete?: () => Promise<void>;
  login?: (user: TPostAuthLoginDTO) => Promise<void>;
  setup?: () => Promise<void>;
  register?: () => Promise<void>;
}

export interface IPromptEvent extends Event {
  prompt: () => Promise<void>;
}

export interface IGetLatestVersionDTO {
  'dist-tags': { latest: string };
}
