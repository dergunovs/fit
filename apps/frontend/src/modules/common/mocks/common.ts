import { vi } from 'vitest';

export const spyCopyToClipboard = vi.spyOn(navigator.clipboard, 'writeText').mockResolvedValue();
