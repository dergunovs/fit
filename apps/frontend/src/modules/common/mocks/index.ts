export * from './helpers';
export * from './ui';
export * from './router';
export * from './layout';
export * from './pwa';
export * from './service';
