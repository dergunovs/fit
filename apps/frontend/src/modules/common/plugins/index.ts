import { createI18n, I18nOptions } from 'vue-i18n';

import ru from '@/common/locales/ru.json';
import en from '@/common/locales/en.json';

const messages = { en, ru } as const;

const options: I18nOptions = { legacy: false, locale: 'ru', messages };

// eslint-disable-next-line @typescript-eslint/no-type-alias
export type TMessageSchema = typeof en;

export const i18n = createI18n<false, typeof options>(options);
