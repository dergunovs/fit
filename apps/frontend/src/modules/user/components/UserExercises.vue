<template>
  <UiFlex column gap="16">
    <UiFlex wrap>
      <UiChip v-for="exercise in props.exercises" :key="exercise._id" align="center" data-test="user-exercises">
        <span data-test="user-exercise-title">{{ exercise[localeField('title', locale)] }}</span>

        <button
          v-if="exercise._id"
          @click="emit('edit', exercise)"
          type="button"
          :class="$style.edit"
          data-test="user-exercise-edit"
        >
          <IconEdit width="20" height="20" />
        </button>
      </UiChip>
    </UiFlex>
  </UiFlex>
</template>

<script setup lang="ts">
import { useI18n } from 'vue-i18n';
import { UiFlex, UiChip } from 'mhz-ui';
import { localeField } from 'mhz-helpers';
import { IExercise } from 'fitness-tracker-contracts';

import IconEdit from '@/common/icons/edit.svg';

interface IProps {
  exercises?: IExercise[];
}

interface IEmit {
  edit: [exercise: IExercise];
}

const props = defineProps<IProps>();
const emit = defineEmits<IEmit>();

const { locale } = useI18n();
</script>

<style module lang="scss">
.edit {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2px;
  cursor: pointer;
  background: none;
  border: none;
}
</style>
