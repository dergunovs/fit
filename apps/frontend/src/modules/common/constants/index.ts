import IconExercise from '@/common/icons/nav-exercise.svg?component';
import IconActivity from '@/common/icons/nav-activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconUser from '@/common/icons/nav-user.svg?component';

import { URL_USER } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { INavItem } from '@/common/interface';
import { URL_SETUP } from '@/auth/constants';

export const NAV_ITEMS: INavItem[] = [
  { _id: '1', url: URL_EXERCISE, title: 'Упражнения', icon: IconExercise },
  { _id: '2', url: URL_ACTIVITY_ADMIN, title: 'Занятия', icon: IconActivity },
  { _id: '3', url: URL_EQUIPMENT, title: 'Оборудование', icon: IconEquipment },
  { _id: '4', url: URL_USER, title: 'Пользователи', icon: IconUser },
];

export const UPDATE_BUTTON_TEXT = 'Обновить';
export const CREATE_BUTTON_TEXT = 'Создать';

export const URL_HOME = '/';
export const URL_ERROR = '/404';

export const GUEST_PAGES = [URL_HOME, URL_SETUP];
