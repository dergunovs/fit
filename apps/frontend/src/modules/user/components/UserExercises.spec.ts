import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import UserExercises from './UserExercises.vue';

import { wrapperFactory } from '@/common/test';
import { EXERCISES_FIXTURE } from '@/exercise/fixtures';

const userExercises = dataTest('user-exercises');
const userExerciseTitle = dataTest('user-exercise-title');
const userExerciseEdit = dataTest('user-exercise-edit');

let wrapper: VueWrapper<InstanceType<typeof UserExercises>>;

beforeEach(() => {
  wrapper = wrapperFactory(UserExercises, { exercises: EXERCISES_FIXTURE });
});

enableAutoUnmount(afterEach);

describe('UserExercises', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UserExercises)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows user exercises', async () => {
    expect(wrapper.findAll(userExercises).length).toStrictEqual(EXERCISES_FIXTURE.length);
    expect(wrapper.find(userExerciseTitle).text()).toStrictEqual(EXERCISES_FIXTURE[0].title);
  });

  it('emits user exercise to edit', async () => {
    expect(wrapper.emitted()).not.toHaveProperty('edit');

    await wrapper.find(userExerciseEdit).trigger('click');

    expect(wrapper.emitted('edit')).toHaveLength(1);
    expect(wrapper.emitted()['edit'][0]).toStrictEqual([EXERCISES_FIXTURE[0]]);
  });
});
