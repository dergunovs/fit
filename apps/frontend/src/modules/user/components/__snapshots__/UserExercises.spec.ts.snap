// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserExercises > matches snapshot 1`] = `
"<div column="" gap="16">
  <h3>Ваши упражнения</h3>
  <div wrap="">
    <div align="center" data-test="user-exercises"><span data-test="user-exercise-title">Отжимание от скамьи</span><button type="button" class="_edit_29d537" data-test="user-exercise-edit">
        <anonymous-stub width="20" height="20"></anonymous-stub>
      </button></div>
    <div align="center" data-test="user-exercises"><span data-test="user-exercise-title">Подтягивание</span><button type="button" class="_edit_29d537" data-test="user-exercise-edit">
        <anonymous-stub width="20" height="20"></anonymous-stub>
      </button></div>
    <div align="center" data-test="user-exercises"><span data-test="user-exercise-title">Жим лёжа</span><button type="button" class="_edit_29d537" data-test="user-exercise-edit">
        <anonymous-stub width="20" height="20"></anonymous-stub>
      </button></div>
  </div>
</div>"
`;
